tests/cache_sorted_set.rs

```diff
@@ -10,40 +12,26 @@ use momento::requests::cache::sorted_set::sorted_set_fetch_response::SortedSetFe
 use momento_test_util::CACHE_TEST_STATE;
 
 #[tokio::test]
-async fn sorted_set_put_element_happy_path() {
-    let client = CACHE_TEST_STATE.client.clone();
-    let cache_name = CACHE_TEST_STATE.cache_name.clone();
+async fn sorted_set_put_element_happy_path() -> MomentoResult<()> {
+    let client = &CACHE_TEST_STATE.client;
+    let cache_name = &CACHE_TEST_STATE.cache_name;
     let sorted_set_name = "sorted-set-".to_string() + &Uuid::new_v4().to_string();
     let value = "value";
     let score = 1.0;
 
     let result = client
-        .sorted_set_fetch_by_rank(
-            cache_name.clone(),
-            sorted_set_name.clone(),
-            Ascending,
-            None,
-            None,
-        )
-        .await
-        .unwrap();
+        .sorted_set_fetch_by_rank(cache_name, &*sorted_set_name, Ascending, None, None)
+        .await?;
     assert_eq!(result, SortedSetFetch::Miss);
 
     client
-        .sorted_set_put_element(cache_name.clone(), sorted_set_name.clone(), "value", 1.0)
+        .sorted_set_put_element(cache_name, &*sorted_set_name, "value", 1.0)
```

Comment:
not relevant to this PR but I really want to understand this; why is the `&*` required, why doesn't just `&` work? would you mind capturing that in a separate ticket?

Reply:
I think it's to dereference it to get the underlying type, then create a reference from that underlying type: https://stackoverflow.com/questions/41273041/what-does-combined-together-do-in-rust